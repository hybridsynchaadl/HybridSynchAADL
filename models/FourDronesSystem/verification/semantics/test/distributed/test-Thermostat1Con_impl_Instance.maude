set show timing off .

load ../../interpreter-distributed.maude
load ../syntax/Thermostat1Con_impl_Instance.maude .

mod TEST-THERMOSTAT1CON-SYSTEM is
  including THERMOSTAT1CON_IMPL_INSTANCE-MODEL .
  including MODEL-TRANSITION-SYSTEM .

  op initState : -> Object .
  eq initState = initialize(initial) .

  op cinitState : -> Object .
  eq cinitState = initialize(collapse(initial)) .

  eq #sample = 4 .
  eq #response = 4 .
endm

--- set print attribute on .

rew  [100] {initState | 0 | infty} .

rew  [100] {cinitState | 0 | infty} .

--- the number of reachable states up to 2 synchronous step
search [,5] {initState | 0 | infty} =>* {none | T:Time | infty} .

search [,5] {cinitState | 0 | infty} =>* {none | T:Time | infty} .
