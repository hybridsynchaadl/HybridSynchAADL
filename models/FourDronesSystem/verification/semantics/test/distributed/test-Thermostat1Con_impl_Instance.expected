==========================================
rewrite [100] in TEST-THERMOSTAT1CON-SYSTEM : {initState | 0 | infty} .
rewrites: 11026
result GlobalSystem: {< Thermostat1ConimplInstance : System | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : none,subcomponents : (< console : System | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : (< tempmax : AsyncDataOutPort | properties :
    DataModel::InitialValue => {{30}},content : [0.0] # [false],destination : (
    controller .. tempmax) > < tempmin : AsyncDataOutPort | properties :
    DataModel::InitialValue => {{10}},content : [0.0] # [false],destination : (
    controller .. tempmin) >),subcomponents : < consoleProc : Process |
    properties : ((TimingProperties::Period => {{5}}) ;
    HybridSynchAADL::Synchronous => {{true}}),features : (< tempmax :
    AsyncDataOutPort | properties : none,content : [0.0] # [false],destination
    : tempmax > < tempmin : AsyncDataOutPort | properties : none,content : [
    0.0] # [false],destination : tempmin >),subcomponents : < consoleThread :
    AsyncThread | properties : ((TimingProperties::Period => {{5}}) ;
    HybridSynchAADL::Synchronous => {{true}}),features : (< tempmax :
    AsyncDataOutPort | properties : none,content : [0.0] # [false],destination
    : tempmax > < tempmin : AsyncDataOutPort | properties : none,content : [
    0.0] # [false],destination : tempmin >),subcomponents : none,connections :
    empty,period : 5,perTimer : 5,msgTimer : infty,varGen : <
    "Thermostat1ConimplInstance.console.consoleProc.consoleThread" >,currState
    : idle,completeStates : idle,variables : empty,transitions : ((select -[[[
    true]]]-> idle {(f{tempmax} := [[30]]) ; f{tempmin} := [[10]]}) ; idle -[on
    dispatch]-> select {skip}),loopBound : 10,transBound : 10,execTime : 0,
    execTimer : infty > > > < controller : System | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; (HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}})
    ; (HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< tempmax :
    AsyncDataInPort | properties : none,content : [0.0] # [false],destination :
    (ctrlProc .. tempmax),cache : [0.0] # [false] > < oncontrol :
    AsyncDataOutPort | properties : none,content : * # [false],destination : (
    environment .. oncontrol) > < currtemp : AsyncDataInPort | properties :
    DataModel::InitialValue => {{15}},content : [0.0] # [false],destination : (
    ctrlProc .. currtemp),cache : [0.0] # [false] > < offcontrol :
    AsyncDataOutPort | properties : none,content : * # [false],destination : (
    environment .. offcontrol) > < tempmin : AsyncDataInPort | properties :
    none,content : [0.0] # [false],destination : (ctrlProc .. tempmin),cache :
    [0.0] # [false] >),subcomponents : < ctrlProc : Process | properties : ((
    TimingProperties::Period => {{5}}) ; HybridSynchAADL::Synchronous => {{
    true}}),features : (< tempmax : AsyncDataInPort | properties : none,content
    : [0.0] # [false],destination : (ctrlThread .. tempmax),cache : [0.0] # [
    false] > < oncontrol : AsyncDataOutPort | properties : none,content : * # [
    false],destination : oncontrol > < currtemp : AsyncDataInPort | properties
    : none,content : [0.0] # [false],destination : (ctrlThread .. currtemp),
    cache : [0.0] # [false] > < offcontrol : AsyncDataOutPort | properties :
    none,content : * # [false],destination : offcontrol > < tempmin :
    AsyncDataInPort | properties : none,content : [0.0] # [false],destination :
    (ctrlThread .. tempmin),cache : [0.0] # [false] >),subcomponents : <
    ctrlThread : AsyncThread | properties : ((TimingProperties::Period => {{
    5}}) ; (HybridSynchAADL::Synchronous => {{true}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< tempmax :
    AsyncDataInPort | properties : none,content : [3.0e+1] # [true],destination
    : empty,cache : [3.0e+1] # [true] > < oncontrol : AsyncDataOutPort |
    properties : none,content : * # [false],destination : oncontrol > <
    currtemp : AsyncDataInPort | properties : none,content : [0.0] # [false],
    destination : empty,cache : [2.3125] # [true] > < offcontrol :
    AsyncDataOutPort | properties : none,content : * # [false],destination :
    offcontrol > < tempmin : AsyncDataInPort | properties : none,content : [
    1.0e+1] # [true],destination : empty,cache : [1.0e+1] # [true] >),
    subcomponents : none,connections : empty,period : 5,perTimer : 5,msgTimer :
    infty,varGen : <
    "Thermostat1ConimplInstance.controller.ctrlProc.ctrlThread" >,currState :
    init,completeStates : init,variables : empty,transitions : init -[on
    dispatch]-> init {if(f[currtemp] > f[tempmax]){offcontrol !} (elsif(f[
    currtemp] < f[tempmin]){oncontrol !}) end if},loopBound : 10,transBound :
    10,execTime : 5/2,execTimer : 5/2 > > > < environment : AsyncEnv |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::Synchronous => {{true}}) ;
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),features :
    (< oncontrol : AsyncEnvInPort | properties : none,content : * # [true],
    envCache : null(Unit),target : controller,destination : empty > < temp :
    AsyncEnvOutPort | properties : none,content : [0.0] # [false],envCache :
    null(Real),target : controller,destination : (controller .. currtemp) > <
    offcontrol : AsyncEnvInPort | properties : none,content : * # [false],
    envCache : null(Unit),target : controller,destination : empty >),
    subcomponents : < x : Data | properties : DataModel::InitialValue => {{
    16}},features : none,subcomponents : none,connections : empty,value : [
    9.4375] # [true] >,connections : x ==> temp,period : 5,perTimer : 5,
    msgTimer : infty,varGen : < "Thermostat1ConimplInstance.environment" >,
    currMode : on,jumps : ((off -[oncontrol]-> on) ; (off -[offcontrol]-> off)
    ; (on -[oncontrol]-> on) ; on -[offcontrol]-> off),flows : ((off [x(t)= (c[
    x] + (minus([[3.0]]) * v[t]))]) ; on [x(t)= (c[x] + ([[1.5]] * v[t]))]),
    sampling : controller :(0,2),response : controller :(3,4),samplingTimes :
    controller : (1/4, 3/4, 5/4, 7/4),responseTimes : controller : (25/8, 27/8,
    29/8, 31/8),envState : ([sampling,controller] [response,controller,25/8]),
    envTimer : 1/4 >) > | 20 | infty}
==========================================
rewrite [100] in TEST-THERMOSTAT1CON-SYSTEM : {cinitState | 0 | infty} .
rewrites: 8668
result GlobalSystem: {< Thermostat1ConimplInstance : System | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : none,subcomponents : (< environment : AsyncEnv | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : (< oncontrol : AsyncEnvInPort | properties : none,content : * #
    [true],envCache : null(Unit),target : controller,destination : empty > <
    temp : AsyncEnvOutPort | properties : none,content : [0.0] # [false],
    envCache : null(Real),target : controller,destination : ((controller .
    ctrlProc . ctrlThread) .. currtemp) > < offcontrol : AsyncEnvInPort |
    properties : none,content : * # [false],envCache : null(Unit),target :
    controller,destination : empty >),subcomponents : < x : Data | properties :
    DataModel::InitialValue => {{16}},features : none,subcomponents : none,
    connections : empty,value : [9.4375] # [true] >,connections : x ==> temp,
    period : 5,perTimer : 5,msgTimer : infty,varGen : <
    "Thermostat1ConimplInstance.environment" >,currMode : on,jumps : ((off -[
    oncontrol]-> on) ; (off -[offcontrol]-> off) ; (on -[oncontrol]-> on) ; on
    -[offcontrol]-> off),flows : ((off [x(t)= (c[x] + (minus([[3.0]]) * v[
    t]))]) ; on [x(t)= (c[x] + ([[1.5]] * v[t]))]),sampling : controller :(0,
    2),response : controller :(3,4),samplingTimes : controller : (1/4, 3/4,
    5/4, 7/4),responseTimes : controller : (25/8, 27/8, 29/8, 31/8),envState :
    ([sampling,controller] [response,controller,25/8]),envTimer : 1/4 > <
    console . consoleProc . consoleThread : AsyncThread | properties : ((
    TimingProperties::Period => {{5}}) ; (HybridSynchAADL::Synchronous => {{
    true}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : (< tempmax : AsyncDataOutPort | properties :
    DataModel::InitialValue => {{30}},content : [0.0] # [false],destination : (
    (controller . ctrlProc . ctrlThread) .. tempmax) > < tempmin :
    AsyncDataOutPort | properties : DataModel::InitialValue => {{10}},content :
    [0.0] # [false],destination : ((controller . ctrlProc . ctrlThread) ..
    tempmin) >),subcomponents : none,connections : empty,period : 5,perTimer :
    5,msgTimer : infty,varGen : <
    "Thermostat1ConimplInstance.console.consoleProc.consoleThread" >,currState
    : idle,completeStates : idle,variables : empty,transitions : ((select -[[[
    true]]]-> idle {(f{tempmax} := [[30]]) ; f{tempmin} := [[10]]}) ; idle -[on
    dispatch]-> select {skip}),loopBound : 10,transBound : 10,execTime : 0,
    execTimer : infty > < controller . ctrlProc . ctrlThread : AsyncThread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::Synchronous => {{true}}) ; (
    HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< tempmax :
    AsyncDataInPort | properties : none,content : [3.0e+1] # [true],destination
    : empty,cache : [3.0e+1] # [true] > < oncontrol : AsyncDataOutPort |
    properties : none,content : * # [false],destination : (environment ..
    oncontrol) > < currtemp : AsyncDataInPort | properties :
    DataModel::InitialValue => {{15}},content : [0.0] # [false],destination :
    empty,cache : [2.3125] # [true] > < offcontrol : AsyncDataOutPort |
    properties : none,content : * # [false],destination : (environment ..
    offcontrol) > < tempmin : AsyncDataInPort | properties : none,content : [
    1.0e+1] # [true],destination : empty,cache : [1.0e+1] # [true] >),
    subcomponents : none,connections : empty,period : 5,perTimer : 5,msgTimer :
    infty,varGen : <
    "Thermostat1ConimplInstance.controller.ctrlProc.ctrlThread" >,currState :
    init,completeStates : init,variables : empty,transitions : init -[on
    dispatch]-> init {if(f[currtemp] > f[tempmax]){offcontrol !} (elsif(f[
    currtemp] < f[tempmin]){oncontrol !}) end if},loopBound : 10,transBound :
    10,execTime : 5/2,execTimer : 5/2 >) > | 20 | infty}
==========================================
search [, 5] in TEST-THERMOSTAT1CON-SYSTEM : {initState | 0 | infty} =>* {none
    | T:Time | infty} .

No solution.
states: 2155  rewrites: 150899
==========================================
search [, 5] in TEST-THERMOSTAT1CON-SYSTEM : {cinitState | 0 | infty} =>* {none
    | T:Time | infty} .

No solution.
states: 2155  rewrites: 127091
Bye.
